x-pack/plugins/lists/public/exceptions/hooks/use_exception_list.tsx

```diff
@@ -37,64 +48,60 @@ export const useExceptionList = ({
   },
   onError,
 }: UseExceptionListProps): ReturnExceptionListAndItems => {
-  const [exceptionListAndItems, setExceptionList] = useState<ExceptionListAndItems | null>(null);
-  const [shouldRefresh, setRefresh] = useState<boolean>(true);
-  const refreshExceptionList = useCallback(() => setRefresh(true), [setRefresh]);
+  const [exceptionList, setExceptionList] = useState<ExceptionListSchema | null>(null);
+  const [exceptionItems, setExceptionListItems] = useState<ExceptionItemsAndPagination | null>(
+    null
+  );
+  const fetchExceptionList = useRef<Func | null>(null);
```

Comment:
Just curious here, what was the issue with the `refreshExceptionList  = useCallback(...)` method of refreshing? Were you losing the correct reference to `fetchData` between `useEffect()` ticks? Either way, this appears to be the most common/easy to use pattern we have in the app for triggering a refresh via hooks. This just reminds me that the `useSiemJobs` hook could use some updating from a `refreshToggle` to the 21st century... 😅

Reply:
No so it was actually preventing some wanted functionality. With the hook, the list and items get called any tie the id and namespace change which is what we want, but the api call would only occur once because after that first successful call the `shouldRefresh` boolean was set to `false`. Ended up having to manually call to fetch list on id change which defeated the purpose.